src/vs/vscode.d.ts

```diff
@@ -4498,6 +4498,16 @@ declare module 'vscode' {
 		export function getLanguages(): Thenable<string[]>;
 
 		/**
+		 * Get the default language associated with a uri.
+		 *
+		 * @param uri The uri of the resource.
+		 * @param uri A location within the resource to check. Used for embedded languages.
+
+		 * @return Promise resolving to language identifier.
+		 */
+		export function getLanguage(uri: Uri, position?: Position): Thenable<string>;
```

Comment:
Strictly speaking we should offer synchronous reading API, we already messed up with `getLanguages()` on this unsure if we should continue like that.

The position-thing touches on @ramya-rao-a requirements of knowing what languages are embedded. So there is overlap with a request to expose token information and I am not sure it's the right thing to mix things here. Especially because the "outer" mode we know with higher certainty than the "inner" mode. I mean the outer mode is determined via the filename or contents whereas the inner mode requires flaky, imprecise grammars to execute.